contrib/DEVELOPMENT_WORKFLOW.md

```diff
@@ -1,63 +1,410 @@
+
+#### License
+---------------
+
+The script is licensed under the MIT License. The copyright is held by Yuma Rao. The license allows anyone to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the software, subject to certain conditions.
+
+#### Usage
+-------------
+
+The `metagraph` class can be used to maintain and manipulate the state of a chain in a PyTorch model. It can be saved to a file and loaded from a file for persistence. It can also be synced with the chain to update its state.
+
+### 3. Subtensor
+
+This Python script is part of a blockchain system, specifically designed to handle various data structures such as `NeuronInfo`, `NeuronInfoLite`, `PrometheusInfo`, `DelegateInfo`, and `SubnetInfo`. These data classes represent different types of information in the system, including neuron metadata, delegate info, and subnet info.
+
+The script also includes methods for encoding and decoding these data structures to and from a serialized format using SCALE encoding, a common choice for blockchain systems.
+
+Key components of the script include:
+
+`Custom RPC Type Registry`: This dictionary defines the data structures for various types of information in the system, such as `SubnetInfo`, `DelegateInfo`, `NeuronInfo`, etc. It is used in the decoding process to interpret the serialized data.
+
+`ChainDataType Enum`: This enumeration is used to specify the type of data being handled, which can be one of `NeuronInfo`, `SubnetInfo`, `DelegateInfo`, `NeuronInfoLite`, or `DelegatedInfo`.
+
+`from_scale_encoding Function`: This function is used to decode data from SCALE format. It uses the `scalecodec` library to handle the decoding process. The function takes a list of integers (the serialized data), the type of data, and flags indicating whether the data is a vector or an option. It returns a dictionary representing the decoded data.
+
+`Data Classes`: The script defines several data classes (`NeuronInfo`, `NeuronInfoLite`, `PrometheusInfo`, `DelegateInfo`, `SubnetInfo`) to represent different types of information in the system. Each class includes a `from_vec_u8` method to create an instance of the class from serialized data, and a `fix_decoded_values` method to correct the format of decoded data.
+
+`ProposalVoteData and ProposalCallData`: These are data structures used for handling proposal data in the system. `ProposalVoteData` is a dictionary that includes information about a proposal's votes, while `ProposalCallData` is a generic call object from the `scalecodec` library.
+
+Please note that this script requires the bittensor and scalecodec libraries, which should be installed and properly configured in your environment.
+
+
+Notes for Commit
```

Comment:
This is clearly outlined in the new STYLE.md guide.  Unnecessarily duplicated and would be effective to just refer/link to the guide instead of slightly different guidelines.